src/binder/assemblybinder.cpp

```diff
@@ -689,6 +689,11 @@ namespace BINDER_SPACE
         sCoreLib = sCoreLibDir;
         sCoreLib.Append(CoreLibName_IL_W);
         BOOL fExplicitBindToNativeImage = (fBindToNativeImage == true)? TRUE:FALSE;
+#if defined(FEATURE_PAL) && !defined(_TARGET_AMD64_)      
+        // Non-Amd64 platforms on non-Windows do not support generating the NI image
+        // as CoreLib.dll. For those, we will bind as IL.
```

Comment:
@gkhanna79 Why does this work on AMD64 *NIX?  What is the root cause?  I assume this has something to do with out mmap discussion.